### PR TITLE
The glibc patch for the Dockerfile was only applied to the builder

### LINKED ISSUE #2309: with-docker build failure yarn install
### What version of Turborepo are you using?

latest as of Oct 23rd 10:09 pm est

### What package manager are you using / does the bug impact?

Yarn v1

### What operating system are you using?

Mac (M1 max)

### Describe the Bug

- Trying to use turborepo to deploy to google cloud run
- Using `turborepo/examples/with-docker` as starting point
- cd to `with-docker` root
- Run the following build command: `docker build --progress=plain -t test-turbo-cloud-run-api -f apps/api/Dockerfile .`
  - Same failure occurs when following the steps in `with-docker`s README to the letter.
- Docker build fails at `RUN yarn install` with this error...

```
#21 19.79 [5/5] Building fresh packages...
#21 20.29 error /app/node_modules/turbo: Command failed.
#21 20.29 Exit code: 1
#21 20.29 Command: node install.js
#21 20.29 Arguments:
#21 20.29 Directory: /app/node_modules/turbo
#21 20.29 Output:
#21 20.29 node:internal/errors:863
#21 20.29   const err = new Error(message);
#21 20.29               ^
#21 20.29
#21 20.29 Error: Command failed: node /app/node_modules/turbo/bin/turbo --version
#21 20.29 /app/node_modules/turbo/bin/turbo:13
#21 20.29     throw e;
#21 20.29     ^
#21 20.29
#21 20.29 <ref *1> Error: spawnSync /app/node_modules/turbo-linux-arm64/bin/turbo ENOENT
#21 20.29     at Object.spawnSync (node:internal/child_process:1117:20)
#21 20.29     at spawnSync (node:child_process:857:24)
#21 20.29     at Object.execFileSync (node:child_process:900:15)
#21 20.29     at Object.<anonymous> (/app/node_modules/turbo/bin/turbo:6:28)
#21 20.29     at Module._compile (node:internal/modules/cjs/loader:1159:14)
#21 20.29     at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
#21 20.29     at Module.load (node:internal/modules/cjs/loader:1037:32)
#21 20.29     at Module._load (node:internal/modules/cjs/loader:878:12)
#21 20.29     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:82:12)
#21 20.29     at node:internal/main/run_main_module:23:47 {
#21 20.29   errno: -2,
#21 20.29   code: 'ENOENT',
#21 20.29   syscall: 'spawnSync /app/node_modules/turbo-linux-arm64/bin/turbo',
#21 20.29   path: '/app/node_modules/turbo-linux-arm64/bin/turbo',
#21 20.29   spawnargs: [ '--version' ],
#21 20.29   error: [Circular *1],
#21 20.29   status: null,
#21 20.29   signal: null,
#21 20.29   output: null,
#21 20.29   pid: 67,
#21 20.29   stdout: null,
#21 20.29   stderr: null
#21 20.29 }
#21 20.29
#21 20.29 Node.js v19.0.0
#21 20.29
#21 20.29     at checkExecSyncError (node:child_process:871:11)
#21 20.29     at Object.execFileSync (node:child_process:907:15)
#21 20.29     at validateBinaryVersion (/app/node_modules/turbo/install.js:25:6)
#21 20.29     at /app/node_modules/turbo/install.js:305:5 {
#21 20.29   status: 1,
#21 20.29   signal: null,
#21 20.29   output: [
#21 20.29     null,
#21 20.29     Buffer(0) [Uint8Array] [],
#21 20.29     Buffer(1089) [Uint8Array] [
#21 20.29        47,  97, 112, 112,  47, 110, 111, 100, 101,  95, 109, 111,
#21 20.29       100, 117, 108, 101, 115,  47, 116, 117, 114,  98, 111,  47,
#21 20.29        98, 105, 110,  47, 116, 117, 114,  98, 111,  58,  49,  51,
#21 20.29        10,  32,  32,  32,  32, 116, 104, 114, 111, 119,  32, 101,
#21 20.29        59,  10,  32,  32,  32,  32,  94,  10,  10,  60, 114, 101,
#21 20.29       102,  32,  42,  49,  62,  32,  69, 114, 114, 111, 114,  58,
#21 20.29        32, 115, 112,  97, 119, 110,  83, 121, 110,  99,  32,  47,
#21 20.29        97, 112, 112,  47, 110, 111, 100, 101,  95, 109, 111, 100,
#21 20.29       117, 108, 101, 115,
#21 20.29       ... 989 more items
#21 20.29     ]
#21 20.29   ],
#21 20.29   pid: 49,
#21 20.29   stdout: Buffer(0) [Uint8Array] [],
#21 20.29   stderr: Buffer(1089) [Uint8Array] [
#21 20.29      47,  97, 112, 112,  47, 110, 111, 100, 101,  95, 109, 111,
#21 20.29     100, 117, 108, 101, 115,  47, 116, 117, 114,  98, 111,  47,
#21 20.29      98, 105, 110,  47, 116, 117, 114,  98, 111,  58,  49,  51,
#21 20.29      10,  32,  32,  32,  32, 116, 104, 114, 111, 119,  32, 101,
#21 20.29      59,  10,  32,  32,  32,  32,  94,  10,  10,  60, 114, 101,
#21 20.29     102,  32,  42,  49,  62,  32,  69, 114, 114, 111, 114,  58,
#21 20.29      32, 115, 112,  97, 119, 110,  83, 121, 110,  99,  32,  47,
#21 20.29      97, 112, 112,  47, 110, 111, 100, 101,  95, 109, 111, 100,
#21 20.29     117, 108, 101, 115,
#21 20.29     ... 989 more items
#21 20.29   ]
#21 20.29 }
#21 20.29
#21 20.29 Node.js v19.0.0
#21 20.29 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
#21 ERROR: executor failed running [/bin/sh -c yarn install]: exit code: 1
------
 > [installer  7/10] RUN yarn install:
------
executor failed running [/bin/sh -c yarn install]: exit code: 1
```

### Expected Behavior

Docker container builds successfully, I can deploy to docker hub, then deploy to google cloud run.

### To Reproduce

See the description, it contains repro steps.

### PR DESCRIPTION
**Pull request by rafaeltab**
As the title explains, the Glibc patch for the Dockerfile in the `with-docker` example was only applied to the builder stage and not the installer stage. This caused an error during the docker builds.

I also updated the docs to include the updated Dockerfile.

Closes #2309 and #2286 
PR #2228 previously fixed this for the builder stage, but not for the installer stage

### CODE REVIEW

File: docs/pages/repo/docs/handbook/deploying-with-docker.mdx [hunk 1]
```diff
@@ -112,44 +112,39 @@ Our detailed [`with-docker` example](https://github.com/vercel/turbo/tree/main/e
...
 FROM node:alpine AS runner
 WORKDIR /app
 
 # Don't run production as root
-RUN addgroup --system --gid 1001 nodejs
-RUN adduser --system --uid 1001 nextjs
-USER nextjs
+RUN addgroup --system --gid 1001 expressjs
+RUN adduser --system --uid 1001 expressjs
+USER expressjs
```
Comment (verygoodsoftwarenotvirus) [inline docs/pages/repo/docs/handbook/deploying-with-docker.mdx:146]: this should be USER nextjs, no?

Comment (rafaeltab) [reply docs/pages/repo/docs/handbook/deploying-with-docker.mdx]: It seems I copied over the wrong Dockerfile, I added a comment to the api Dockerfile that tells people to copy the web Dockerfile over.

**Review by chris-olszewski — APPROVED** [review]
Thanks for fixing up the example!

**Comment by verygoodsoftwarenotvirus** [comment]
I can confirm I've applied the core fix made in this change locally and it works. The confusion this caused cost me something like a day and a half of productivity, so for whatever my comment is worth, I think this should be merged.

**Pull request merged**